tests/tests/src/lampo_cln_tests.rs

```diff
@@ -539,3 +539,152 @@ fn be_able_to_kesend_payments() {
+                Event::Lightning(LightningEvent::ChannelReady { .. }) => return Ok(()),
+                _ => continue,
+            };
+        }
+        Err(())
+    });
+
+    wait!(|| {
+        let channels = cln.rpc().listfunds().unwrap().channels;
+        if channels.is_empty() {
+            return Err(());
+        }
+
+        let mut channels = cln.rpc().listfunds().unwrap().channels;
+        let origin_size = channels.len();
+        channels.retain(|chan| chan.state == "CHANNELD_NORMAL");
+        if channels.len() == origin_size {
+            return Ok(());
+        }
+
+        let channels: response::Channels = lampo.call("channels", json::json!({})).unwrap();
+        if !channels.channels.first().unwrap().ready {
+            return Err(());
+        }
+        let address = cln.rpc().newaddr(None).unwrap();
+        fund_wallet(btc.clone(), &address.bech32.unwrap(), 1).unwrap();
+        crate::wait_cln_sync!(cln);
+        Err(())
+    });
+
+    // This should return the final channel_id as channel_id may differ from the time being in ChannelPending, ChannelClosed state.
+    let channels: response::Channels = lampo.call("channels", json::json!({})).unwrap();
+
+    // This should fail as there are two channels with the peer so we need to pass the specific `channel_id`
+    let result: Result<response::CloseChannel, _> = lampo.call(
+        "close",
+        request::CloseChannel {
+            node_id: info_cln.id.to_string(),
+            channel_id: None,
+        },
+    );
+
+    assert!(result.is_err(), "{:?}", result);
+
+    // Closing the first channel
+    let result: Result<response::CloseChannel, _> = lampo.call(
+        "close",
+        request::CloseChannel {
+            node_id: info_cln.id.to_string(),
+            channel_id: Some(channels.channels.first().unwrap().channel_id.to_string()),
+        },
+    );
+    assert!(result.is_ok(), "{:?}", result);
+    // assert_eq!(&result.unwrap().counterparty_node_id, &info_cln.id.to_string());
+    assert_eq!(
+        &result.unwrap().channel_id,
+        &channels.channels.first().unwrap().channel_id.to_string()
+    );
+
+    // Closing the second channel - at this point there is only 1 channel with the peer
+    let result: Result<response::CloseChannel, _> = lampo.call(
+        "close",
+        request::CloseChannel {
+            node_id: info_cln.id.to_string(),
+            channel_id: None,
+        },
+    );
+    assert!(result.is_ok(), "{:?}", result);
+    assert_eq!(
+        result.unwrap().counterparty_node_id,
+        info_cln.id.to_string()
+    );
+
+    // Closing the third channel (this channel does not exist)
+    let result: Result<response::CloseChannel, _> = lampo.call(
+        "close",
+        request::CloseChannel {
+            node_id: info_cln.id.to_string(),
+            channel_id: Some(channels.channels.first().unwrap().channel_id.to_string()),
+        },
+    );
+    assert!(result.is_err(), "{:?}", result);
```

Comment:
nit: this is a poor test, 99% of the code is copied by other test, and your code is not checking nothing import.

You are missing:

- checking that `CloseChannel` contains the information that you need
- add another test where the channel_id is not specified

Reply:
The test where `channel_id` is not specified is already added. 
```rust
    // Closing the second channel - at this point there is only 1 channel with the peer
    let result: Result<response::CloseChannel, _> = lampo.call(
        "close",
        request::CloseChannel {
            node_id: info_cln.id.to_string(),
            channel_id: None,
        },
    );
    assert!(result.is_ok(), "{:?}", result);
    assert_eq!(
        result.unwrap().peer_id,
        info_cln.id.to_string()
    );
```

Reply:
So I do not understand the test, why you are calling 3 time close channel?

For clarity every time that you close a channel you make a new test, and add the reason inside the name. E.g: `test_lampo_to_cln_close_channel_without_channel_id_success` and so one all the combination that you added

Reply:
There are four tests in a same one. I'll describe each one of them - 
Prior to the test we open two channels.
- 1st test - We call the `close` with a `None` `channel_id`, this adds the purpose that when there are two channels open with the same peer, we have to provide respective `channel_id`.
- 2nd test - Closing the first channel - This test demonstrate that we can close the channel with given `channel_id`
- 3rd test - Now we have only one channel left - We can close this channel without providing `channel_id`.
- 4th test - Now we do not have any channel left - Now closing the channel would give an error.

I avoided making 4 sets of different tests as it would add much more boilerplate code. Let me know if I should make different test cases for each.

Reply:
>Let me know if I should make different test cases for each.

Yes please :)